### ISSUE TITLE
Poetry from within a poetry virtual environment?

### ISSUE THREAD

**Issue by Pluckerpluck** [sdispater/poetry#909]
Quick system rundown:

* Windows 10 Enterprise
* Python 3.7.2
* Pip 18.1
* Poetry 0.12.11

Same versions used within the virtual environment.

-------

This might be related to https://github.com/sdispater/poetry/issues/281 or https://github.com/sdispater/poetry/issues/527

But it looks like that was meant to be solved in 0.12.4? And I'm running 0.12.11 right now (on Windows, Python 3.7).
Further, I am using `.venv` folders in the project directory (so I can get proper working code completion automatically in vscode).
Finally, I installed poetry to the system using `pip`. It's one of the very few things installed globally, and nothing has been installed since (unless poetry itself went rogue).

Anyway, due to some unique situation, I'm trying to create a tool that wraps poetry. Simple enough, except that I want to build the tool itself using poetry...

So currently when I want to test this I `poetry shell` into the environment of the tool I'm creating, I change to a new project directory (with its own pyproject.toml), and then try to run my tool using `python -m my_tool`.

This runs the tool fine, but within that tool, my script runs `poetry install`, because, as I said I'm trying to wrap poetry. 
As long as VIRTUAL_ENV is not set, it will at least attempt to create a virtual environment for the project.
However, for whatever reason, the .venv folder it creates is not a valid virtualenv. As a result, it will end up installing the requirements of the projects in the top level (i.e. my tool's) virtual environment instead.

This following is what can be found in the `.venv/Scripts` directory it attempts to generate:

```powershell
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       21/02/2019     14:13           2299 activate
-a----       21/02/2019     14:13           1039 activate.bat
-a----       21/02/2019     14:13           1510 Activate.ps1
-a----       21/02/2019     14:13            368 deactivate.bat
-a----       21/02/2019     14:13         522768 python.exe
-a----       21/02/2019     14:13         522256 pythonw.exe
```

Any advice on how I can proceed here? I'm happy to make local changes to poetry for my niche use case here, just so I can get this working without having to test it by actually installing it into a VM...

-------

Note: if I create the virtual environment using my system install of poetry, and **then** shell into my tools project, the internal poetry will correctly use the virtual environment.

It just seems to be unable to generate a valid one itself. So I can actually test most things, besides the initial `poetry install`.

**Comment by Pluckerpluck** [comment]
Ok. This is a problem with EnvBuilder not actually installing pip into the virtual environment for some reason. It just fails silently.

Specifically, when `python -Im ensurepip --upgrade --default-pip` is run from within the second venv, it determines that pip is already installed because it finds them in the first venv.

Anyone know anything about this? Does this present a problem if you install poetry using pipx (as that puts it in a virtual environment)? If not, how does poetry avoid this issue?

**Comment by Pluckerpluck** [comment]
https://bugs.python.org/issue35872

Found the issue. It's a bug in Python 3.7.2. It will be fixed in 3.7.3
